### DISCUSSION TITLE
Bookkeeper pod memory consumption

### DISCUSSION THREAD

**Discussion by ghost** [apache/pulsar#22012] [Q&A] [answered]
I'm running some benchmarks on Pulsar deployed on Kubernetes.

After running a benchmark of 2 minutes duration, I've noticed that the memory consumption of the Bookkeeper pods are not going down and staying high indefinitely (partitioned topic, shared subscription).

I checked that all messages are getting properly acknowledged, so I'm wondering what's taking up all the memory.

<img width="1410" alt="image" src="https://github.com/apache/pulsar/assets/115536533/930021bf-d618-4d4a-bd0e-cb19dce6e2b4">

**Comment by lhotari** [comment, marked as answer]
@kunwyoo This is the way Java works. You could switch to use ZGC and configure `SoftMaxHeapSize`. explained here: https://malloc.se/blog/zgc-softmaxheapsize . This will help ZGC to "uncommit" memory. The default delay is 300 seconds, but this can be tuned.

**Reply by ghost** [reply]
Thanks for your answer!
